push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/Sender.java

```diff
@@ -0,0 +1,84 @@
+package org.jboss.aerogear.unifiedpush.message;
+
+import java.util.Collection;
+
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.jboss.aerogear.unifiedpush.api.PushMessageInformation;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+import org.jboss.aerogear.unifiedpush.api.VariantMetricInformation;
+import org.jboss.aerogear.unifiedpush.message.jms.Dequeue;
+import org.jboss.aerogear.unifiedpush.message.jms.DispatchToQueue;
+import org.jboss.aerogear.unifiedpush.message.sender.NotificationSenderCallback;
+import org.jboss.aerogear.unifiedpush.message.sender.PushNotificationSender;
+import org.jboss.aerogear.unifiedpush.message.sender.SenderTypeLiteral;
+import org.jboss.aerogear.unifiedpush.utils.AeroGearLogger;
+
+public class Sender {
+
+    private final AeroGearLogger logger = AeroGearLogger.getInstance(Sender.class);
+
+    @Inject
+    @Any
+    private Instance<PushNotificationSender> senders;
+
+    @Inject
+    @DispatchToQueue
+    private Event<VariantMetricInformation> dispatchVariantMetricEvent;
+
+    public void sendMessagesToPushNetwork(@Observes @Dequeue MessageWithTokens msg) {
+        final Variant variant = msg.getVariant();
+        final UnifiedPushMessage message = msg.getUnifiedPushMessage();
+        final Collection<String> deviceTokens = msg.getDeviceTokens();
+
+        logger.fine("Received message from queue: " + msg.getUnifiedPushMessage().getMessage().getAlert());
+
+        senders.select(new SenderTypeLiteral(variant.getClass())).get()
+                            .sendPushMessage(variant, deviceTokens, message, new SenderServiceCallback(variant, deviceTokens.size(), msg.getPushMessageInformation()));
```

Comment:
This will be addressed in these issues:
https://issues.jboss.org/browse/AGPUSH-1365
https://issues.jboss.org/browse/AGPUSH-1357